presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryPagesStore.java

```diff
@@ -98,13 +101,17 @@ public synchronized void add(Long tableId, Page page)
         boolean done = false;
         long totalRows = 0;
         for (int i = partNumber; i < tableData.getPages().size() && !done; i += totalParts) {
-            Page page = tableData.getPages().get(i);
+            if (sampleRatio.isPresent() && ThreadLocalRandom.current().nextDouble() >= sampleRatio.getAsDouble()) {
```

Comment:
This needs to be `< sampleRatio`. E.g., if sample rate is 90%, only in 10% of the cases a uniform random between [0, 1] will be higher than 0.9.

Also, since we're sampling pages, we shouldn't sample splits (in MemorySplitManager). Otherwise, we'll end up with rate^2 sampling rate (50% of splits, 50% of pages per split ~ 25% of rows overall)

Reply:
Actually, nevermind. This is the condition for skipping the page (`continue`, below), so it's correct.